src/Domain/Task/Handler/RescheduleHandler.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace AppBundle\Domain\Task\Handler;
+
+use AppBundle\Domain\Task\Command\Reschedule;
+use AppBundle\Domain\Task\Event;
+use AppBundle\Entity\Task;
+use SimpleBus\Message\Recorder\RecordsMessages;
+
+class RescheduleHandler
+{
+
+    public function __construct(private RecordsMessages $eventRecorder)
+    { }
+
+    public function __invoke(Reschedule $command)
+    {
+        $task = $command->getTask();
+        $rescheduleDateTime = $command->getRescheduleDateTime();
+
+        $this->eventRecorder->record(new Event\TaskRescheduled($task, $rescheduleDateTime));
+
+        $task->setDoneBefore($rescheduleDateTime);
+        $task->setMetadata('rescheduled', true);
```

Comment:
i'm unsure about the metadata usage, maybe using a `Tags` "rescheduled" on the `Task` is better ? @alexsegura

Reply:
I'm also unsure if `setDoneBefore()` is the right attribute to update, seems like is work from my first tries.